### ISSUE TITLE
WCAG 4.1.2 A: Element with aria-hidden attribute contains focusable descendants (Issue 33)

### ISSUE THREAD

**Issue by CJE001** [civictheme/uikit#360]
### Summary

Via Vision Australia assessment: August 2024

Impact: high

Note: DTA have a 90 day remediation period to address the identified issues within the audit, all issues must be resolved to obtain WCAG 2.2 certification for digital.gov.au

### Steps to reproduce

Global issue – Left side navigation 
[Digital Experience | digital.gov.au](https://www.digital.gov.au/policy/digital-experience) (https://www.digital.gov.au/policy/digital-experience)

### Observed outcome

The third-level sub-navigation links within the side navigation have no accessible name or role. This is because the parent <div> element has been given the aria-hidden="true" attribute. 

![Picture11](https://github.com/user-attachments/assets/20d63f75-a963-483f-94b6-5e77e659f984)

### Code used

```
<ul class="ct-menu ct-menu__sub-menu ct-menu--level-1">
<li class="ct-menu__item ct-menu__item--level-1 ct-menu__item--has-children " data-collapsible="true" data-collapsible-duration="0" data-collapsible-collapsed="" aria-expanded="false">
<a class="ct-link ct-theme-light ct-menu__item__link" href="/policy/digital-experience/digital-service-standard" title="Digital Service Standard" aria-expanded="false">
Digital Service Standard
</a>
<a href="#" class="ct-menu__item__link-trigger" data-collapsible-trigger="" aria-expanded="false" title="Expand Digital Service Standard menu">
<svg class="ct-icon ct-collapsible__icon" width="24" height="24" aria-hidden="true" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
…
</svg>
</a>
<div class="ct-menu__sub-menu__wrapper ct-menu__sub-menu__wrapper--level-2" data-collapsible-panel="" aria-hidden="true">
<ul class="ct-menu ct-menu__sub-menu ct-menu--level-2">
<li class="ct-menu__item ct-menu__item--level-2 ">
<a class="ct-link ct-theme-light ct-menu__item__link" href="/policy/digital-experience/digital-service-standard/services" title="Services covered by the Digital Service Standard">
Services covered by the Digital Service Standard
</a>
</li>
<li class="ct-menu__item ct-menu__item--level-2 ">
<a class="ct-link ct-theme-light ct-menu__item__link" href="/policy/digital-experience/digital-service-standard/dss-transition" title="Transition approach">
Transition approach
</a>
</li>
... 
</ul>
</div>
…
</li>
</ul>
```

### Why this matters

The aria-hidden attribute removes an element and its descendants from the accessibility tree, making the content unavailable to assistive technologies. However, the content remains visible and focusable. 

When a focusable element receives focus, no accessible name or role is announced. Screen readers may announce 'blank' or the page title in an attempt to provide an accessible name. This can result in users having no access to the content and difficulty understanding or using site functionality. 

### Expected outcome

Ensure that the aria-hidden attribute is not used on focusable elements or elements with focusable descendants. 
In this instance, remove the attribute from the <div> element. 

When the sub-navigation lists are visually hidden (collapsed state), use the CSS display:none; property to ensure the links are hidden from all users and do not appear in the sequential keyboard focus order. 

Update this to display:block; when the sub-navigation lists are visible (expanded state). 

### Code example

```
<div class="ct-menu__sub-menu__wrapper ct-menu__sub-menu__wrapper--level-2" data-collapsible-panel="" style="display:none;">
<ul class="ct-menu ct-menu__sub-menu ct-menu--level-2">
<li class="ct-menu__item ct-menu__item--level-2 ">
<a class="ct-link ct-theme-light ct-menu__item__link" href="/policy/digital-experience/digital-service-standard/services" title="Services covered by the Digital Service Standard">
Services covered by the Digital Service Standard
</a>
</li>
<li class="ct-menu__item ct-menu__item--level-2 ">
<a class="ct-link ct-theme-light ct-menu__item__link" href="/policy/digital-experience/digital-service-standard/dss-transition" title="Transition approach">
Transition approach
</a>
</li>
... 
</ul>
</div>
```

### Related issues

For other issues related to this component, please refer to:

•	Issue 10 – Current page shown with presentation alone
•	Issue 22 – Sub-navigation links not keyboard accessible
•	Issue 23 – Illogical focus order when skip link activated
•	Issue 24 – Hidden navigation elements in keyboard focus order
•	Issue 28 – Focused element not visible and has no visible focus indicator
•	Issue 36 – Link has invalid state

**Comment by CJE001** [comment]
**Guideline 4.1: Compatible: _"Maximise compatibility with current and future user agents, including assistive technologies."_**

**4.1.2 Name, Role, Value WCAG 2.0 Level A**

For all user interface components (including but not limited to: form elements, links and components generated by scripts), the name and role can be programmatically determined; states, properties, and values that can be set by the user can be programmatically set; and notification of changes to these items is available to user agents, including assistive technologies.

This Success Criterion is primarily for web developers who create customised interface components. In order to ensure that screen reader users can understand and interact with such components, it is important that web developers communicate their role, status and any other relevant information.

**Comment by alan-cole** [comment]
Work was completed here to add display none to collapsible: https://github.com/civictheme/uikit/pull/462
